### ISSUE TITLE
Lvgl v8

### ISSUE THREAD

**Issue by bosiden** [HX2003/reflow_oven_lvgl_codeblocks_simulator#4]
Hi : )

Do you have any plans on updating lv_settings.c/lv_settings.h to lvgl v8? I think it would make a nice alternative to the current proposed menu here https://github.com/lvgl/lvgl/pull/2603. This version is more appropriate in terms of reduced memory footprint since it only populates setting objects when needed, and free memory when switching pages. 

I spent two days trying to upgrade it but I gave up, I couldn't figure out what is the alternative method of lv_obj_allocate_ext_attr and lv_event_send_func in v8.

**Comment by HX2003** [comment]
I do not have intentions to do that. You see, memory limitations aren't really my concern right now, the current menu widget is an evolution of the original settings _widget_, it became more flexible at the cost of memory usage.

Is the current menu widget using too much ram?

**Comment by bosiden** [comment]
Right, I also think that the new menu widget is much more flexible and would be more suitable for the larger project types.
I ended up re-writing the settings menu to work with V8, it was tedious but its mostly working now. I did that because my project was already heavily based on the settings menu, and it was time effective for me to just upgrade it.

> Is the current menu widget using too much ram?

I seems V8 is using significantly more ram per design, so its not really specific to the menu widget.
